demoapp/app.py

```diff
@@ -1,149 +1,247 @@
-        
-        with col1:
-            st.subheader("Original Bill")
-            st.write(bill_title)
-            st.write(bill_content)
-        
-        with col2:
-            st.subheader("Generated Text")
-            try:
-                update_csv(bill_title, response, csv_file_path)
-                st.write(response)
-                st.download_button(
-                            label="Download Text",
-                            data=pd.read_csv("demoapp/generated_bills.csv").to_csv(index=False).encode('utf-8'),
-                            file_name='Bills_Summarization.csv',
-                            mime='text/csv',)
-        
-            except Exception as e:
-                st.error("no response")
-        
-        with col3:
-            st.subheader("Evaluation Metrics")
-            # st.write(f"Jaccard Score: {calculate_jaccard_similarity(bill_content, response)}")
-            st.write(f"Total Tokens: {total_tokens}")
-            st.write(f"Prompt Tokens: {prompt_tokens}")
-            st.write(f"Completion Tokens: {completion_tokens}")
-            st.write(f"Total Cost (USD): ${total_cost}")
-                
-
-    
+import streamlit as st
+import pandas as pd
+import os
+__import__('pysqlite3')
+import sys
+sys.modules['sqlite3'] = sys.modules.pop('pysqlite3')
+
+from langchain.prompts import PromptTemplate
+from langchain.chains import LLMChain, create_tagging_chain, create_tagging_chain_pydantic
+from langchain.vectorstores import Chroma
+from langchain.chat_models import ChatOpenAI
+from langchain.schema.runnable import RunnablePassthrough
+from langchain.schema.output_parser import StrOutputParser
+from langchain.callbacks import get_openai_callback
+from sklearn.feature_extraction.text import TfidfVectorizer
+from sklearn.metrics.pairwise import cosine_similarity
+from rouge_score import rouge_scorer
+from sentence_transformers import CrossEncoder
+from langchain.embeddings.openai import OpenAIEmbeddings
+from langchain.text_splitter import CharacterTextSplitter
+from langchain.document_loaders import TextLoader
+from sidebar import *
+from tagging import *
+
+
+
+st.set_page_config(page_title="Summarize and Tagging MA Bills", layout='wide')
+st.title('Summarize Bills')
+
+sbar()
+
+
+template = """"You are a summarizer model that summarizes legal bills and legislation. Please include the bill's main purpose, relevant key points and any amendements. 
+The summaries must be easy to understand and accurate based on the provided bill. I want you to summarize the legal bill and legislation. 
+Use the title {title} to guide your summary. Summarize the bill that reads as follows:\n{context}\n\nSummary: An Act [bill title]. This bill [key information].
+"""
+
+# model to test hallucination
+# model = CrossEncoder('vectara/hallucination_evaluation_model')
+
+# load the dataset
+df = pd.read_csv("demoapp/all_bills.csv")
```

Comment:
This file is missing from the demoapp folder, please include a a link to the file or upload it

Reply:
Hi Michelle, the Vectara/hallucination_evaluation_model is to call the model_path url to retrieve the information. We didn't store the model locally.
https://huggingface.co/vectara/hallucination_evaluation_model

Reply:
Apologies for the confusion, I was referring to line 42

Reply:
Hmm, I saw this file exist in the dev branch. Perhaps it is included from previous request?

![pic1](https://github.com/BU-Spark/ml-maple-bill-summarization/assets/84426364/cddd7bec-ee31-4729-89fb-a785e19f577a)

Reply:
yeah all_bills.csv is already in BU Spark's dev branch